cc_plugin_glider/glider_dac.py

```diff
@@ -844,25 +924,13 @@ def check_container_variables(self, dataset):
         }
 
         for container_var in data_struct:
-            test = container_var in dataset.variables
-            if not test:
-                messages.append("Variable %s is missing" %
-                                container_var)
-                continue
-
-            nc_var = dataset.variables[container_var]
-            dtype = np.dtype(data_struct[container_var]['dtype'])
-            test = nc_var.dtype.str == dtype.str
-            score += int(test)
-            if not test:
-                messages.append('%s variable has incorrect dtype, should be %s' % (container_var, dtype.name))  # noqa
-
-            for field in data_struct[container_var]['fields']:
-                test = hasattr(nc_var, field)
-                if not test:
-                    messages.append('%s variable is missing required attribute %s' % (container_var, field))  # noqa
-                    continue
-                score += 1
+            expected_dtype = np.dtype(data_struct[container_var]['dtype'])
+            var_fields = data_struct[container_var]['fields']
```

Comment:
Is there some way to combine data_struct and expected_format.py?

Reply:
Yeah, I'm open to ideas on this.  `expected_format` has attributes as keys and expected values as values.  Sometimes there's custom logic for certain attributes i.e. `units`, which checks if the unit is UDUNITS2 convertible to the specified unit in `expected_format`.  Maybe we could just specify `None` as the value for variables where we just want to check whether a given attribute exists.

Reply:
Upon further examination, it'd be difficult to refactor and keep all the priorities/logic the same.  I'm not saying we shouldn't examine this at some point, but for now there aren't dead code paths, the desired behavior is preserved, and the new checks are implemented.

Reply:
Fair enough, can you add an issue and include your notes from your examination?